mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -309,6 +313,23 @@ private[ml] object DefaultParamsWriter {
     val metadataJson: String = compact(render(metadata))
     metadataJson
   }
+
+  /**
+   * Save estimator's `initialModel` to corresponding path.
+   */
+  def saveInitialModel[T <: HasInitialModel[_ <: MLWritable with Params]](
+      instance: T, path: String): Unit = {
+    if (instance.isDefined(instance.initialModel)) {
+      val initialModelPath = new Path(path, "initialModel").toString
+      val initialModel = instance.getOrDefault(instance.initialModel)
+      // When saving, only keep the direct initialModel by eliminating possible initialModels of the
+      // direct initialModel, to avoid unnecessary deep recursion of initialModel.
+      if (initialModel.hasParam("initialModel")) {
+        initialModel.clear(initialModel.getParam("initialModel"))
+      }
+      initialModel.save(initialModelPath)
```

Comment:
I suggest when setInitialModel, first copy the passed in model (by `Model.copy`) and then clear the `initialModel` param in the passed in model to avoid recursive risk. Because the passed in model maybe used in other place we should avoid to modify its params.

Another way is avoid to add the initial model param to Model, only add it to the Estimator. (This way I think is more reasonable).